int8 low_sensor
int8 up_sensor